packages/cli/src/cli.ts

```diff
@@ -74,6 +74,10 @@ const getConfig = async (configPath = 'tsoa.json'): Promise<Config> => {
   return config;
 };
 
+const resolveConfig = async (config?: string | Config) => {
+  return typeof config === 'object' ? config : await getConfig(config);
```

Comment:
Don't release Zalgo

Reply:
I know about Zalgo but I'm not sure what you're asking for here. This is an async function, and its return value will always be on "a later tick". 

Returning `config` directly will still result in an async function signature / Promise. Adding an await, as in `return await config` *will* further defer by adding an additional Promise resolution, but it doesn't increase determinacy in any way that I can see? For that matter, I guess `getConfig` doesn't need to be awaited, either, and `return (typeof config==="config") ? config : getConfig(config)` would remove an unneeded intermediary Promise resolution.

I added a clarifying return type to the function signature. Let me know if there's something else I am missing here. Thanks for looking this over!

Reply:
Missed the async for some reason, your right